src/conversions/rust_decimal.rs

```diff
@@ -0,0 +1,91 @@
+#![cfg(feature = "rust_decimal")]
+
+use crate::exceptions::PyTypeError;
+use crate::{intern, FromPyObject, IntoPy, PyAny, PyObject, PyResult, Python, ToPyObject};
+use rust_decimal::Decimal;
+
+impl FromPyObject<'_> for Decimal {
+    fn extract(obj: &PyAny) -> PyResult<Self> {
+        // use the string representation to not be lossy
+        // TODO: should this check that the underlying type is of
+        //       decimal.Decimal ?
+        // TODO: should this use obj.call_method0(intern!(py, "__str__"))?
+        // TODO: is PyTypeError the right thing? The Decimal code isn't
+        //       exposed via the C API so its a bit funky to make those
+        //       exceptions directly
+        Decimal::from_str_exact(obj.str()?.to_str()?)
+            .map_err(|e| PyTypeError::new_err(e.to_string()))
+    }
+}
+
+impl ToPyObject for Decimal {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        // TODO: how do we handle the error gracefully?
+        // look up the decimal.Decimal
+        let dec_mod = py
+            .import(intern!(py, "decimal"))
+            .expect("failed to import decimal");
+        let dec_cls = dec_mod
+            .getattr(intern!(py, "Decimal"))
+            .expect("failed getattr Decimal from decimal");
+        // now call the constructor with the Rust Decimal string-ified
+        // to not be lossy
+        // TODO: is to_string() the best conversion?
+        let ret = dec_cls
+            .call1((self.to_string(),))
+            .expect("failed to call decimal.Decimal(value)");
+        ret.to_object(py)
+    }
+}
+
+impl IntoPy<PyObject> for Decimal {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        self.to_object(py)
+    }
+}
+
+#[cfg(test)]
+mod test_rust_decimal {
+    use super::*;
+    use crate::types::PyDict;
+    use indoc::formatdoc;
+    use rust_decimal::Decimal;
+
+    macro_rules! convert_constants {
```

Comment:
It may be interesting to use `proptest` to try conversion of some arbitrary values as a round-trip (you can see examples of this elsewhere in the code).

Reply:
`rust_decimal` didn't support `proptest` so I've now done that at https://github.com/paupino/rust-decimal/pull/582

Reply:
Lifted that into pyo3 so that it would work for now without a bump.